server/aws-lsp-codewhisperer/src/language-server/agenticChat/tools/fuzzySearch.ts

```diff
@@ -23,78 +26,72 @@ export class FileSearch {
-        const writer = updates.getWriter()
-        const closeWriter = async (w: WritableStreamDefaultWriter) => {
-            await w.close()
-            w.releaseLock()
-        }
-        if (!requiresAcceptance) {
-            await writer.write('')
-            await closeWriter(writer)
-            return
+        if (params.threshold) {
+            if (params.threshold > 1 || params.threshold < 0) {
+                throw new Error('Invalid threshold, must be a number between 0 and 1')
+            }
         }
+    }
 
-        const caseSensitiveText = params.caseSensitive ? 'case-sensitive' : 'case-insensitive'
-        if (params.maxDepth === undefined) {
-            await writer.write(
-                `Searching recursively in ${params.path} for files matching pattern "${params.pattern}" (${caseSensitiveText})`
-            )
-        } else if (params.maxDepth === 0) {
-            await writer.write(
-                `Searching in ${params.path} for files matching pattern "${params.pattern}" (${caseSensitiveText})`
-            )
-        } else {
-            const level = params.maxDepth > 1 ? 'levels' : 'level'
-            await writer.write(
-                `Searching in ${params.path} limited to ${params.maxDepth} subfolder ${level} for files matching pattern "${params.pattern}" (${caseSensitiveText})`
-            )
-        }
-        await closeWriter(writer)
+    public async queueDescription(params: FuzzySearchParams, updates: WritableStream, requiresAcceptance: boolean) {
+        // deprecated, no-op
+        return
     }
 
-    public async requiresAcceptance(params: FileSearchParams, approvedPaths?: Set<string>): Promise<CommandValidation> {
+    public async requiresAcceptance(
+        params: FuzzySearchParams,
+        approvedPaths?: Set<string>
+    ): Promise<CommandValidation> {
         return requiresPathAcceptance(params.path, this.workspace, this.logging, approvedPaths)
     }
 
-    public async invoke(params: FileSearchParams): Promise<InvokeOutput> {
+    public async invoke(params: FuzzySearchParams, token?: CancellationToken): Promise<InvokeOutput> {
         const path = sanitize(params.path)
         try {
             // Get all files and directories
             const listing = await workspaceUtils.readDirectoryRecursively(
                 { workspace: this.workspace, logging: this.logging },
                 path,
-                { maxDepth: params.maxDepth, excludeDirs: DEFAULT_EXCLUDE_DIRS, excludeFiles: DEFAULT_EXCLUDE_FILES }
+                { maxDepth: params.maxDepth, excludeDirs: DEFAULT_EXCLUDE_DIRS, excludeFiles: DEFAULT_EXCLUDE_FILES },
+                token
             )
 
-            // Create regex pattern for filtering
-            const regex = new RegExp(params.pattern, params.caseSensitive ? '' : 'i')
+            const fuseOptions = {
+                isCaseSensitive: false,
+                threshold: 0.2,
```

Comment:
any reason to set it as 0.2, I think we do benchmarking with threshold as 0.8.

Reply:
it's reversed in the library: https://www.fusejs.io/api/options.html#threshold